pkg/earlystopping/v1beta1/medianstop/service.py

```diff
@@ -64,14 +64,47 @@ def __init__(self):
 
         self.api_instance = client.CustomObjectsApi()
 
+    def ValidateEarlyStoppingSettings(self, request, context):
+        is_valid, message = self.validate_early_stopping_spec(request.early_stopping_spec)
+        if not is_valid:
+            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+            context.set_details(message)
+            logger.error(message)
+        return api_pb2.ValidateEarlyStoppingSettingsReply()
+
+    def validate_early_stopping_spec(self, early_stopping_spec):
+        algorithm_name = early_stopping_spec.algorithm_name
+        if algorithm_name == "medianstop":
+            return self.validate_medianstop_setting(early_stopping_spec.algorithm_settings)
+        else:
+            return False, "unknown algorithm name {}".format(algorithm_name)
+
+    @staticmethod
+    def validate_medianstop_setting(early_stopping_settings):
+        for setting in early_stopping_settings:
+            try:
+                if setting.name == "min_trials_required":
+                    if not (int(setting.value) >= 0):
+                        return False, "min_trials_required must be greater or equal than zero (>=0)"
```

Comment:
Should we accept only positive numbers ( > 0) to calculate the average ?

Reply:
If users input a negative number or zero, early stopping calculates division by zero in the following code.

https://github.com/kubeflow/katib/blob/16e0574647ace79ccedd248d072c77139feab5e5/pkg/earlystopping/v1beta1/medianstop/service.py#L142

The following is errors output from median-stop service.

```shell
INFO:root:Median stopping settings are: min_trials_required: 0, start_step: 1
ERROR:grpc._server:Exception calling application: division by zero
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/grpc/_server.py", line 434, in _call_behavior
    response_or_iterator = behavior(argument, context)
  File "/opt/katib/pkg/earlystopping/v1beta1/medianstop/service.py", line 93, in GetEarlyStoppingRules
    median = self.getMedianValue(request.trials)
  File "/opt/katib/pkg/earlystopping/v1beta1/medianstop/service.py", line 142, in getMedianValue
    median = sum(list(self.trials_avg_history.values())) / len(self.trials_avg_history)
ZeroDivisionError: division by zero
```

Reply:
So I think my codes are also wrong and should be `if not (int(setting.value) > 0)`.
WDYT? @andreyvelich

Reply:
Sorry, I misunderstood. You were suggesting this.

> Should we accept only positive numbers ( > 0) to calculate the average ?

It makes sense.

Reply:
Yeah, thanks!

Reply:
Done.